src/Paket.Core/Versioning/VersionRange.fs

```diff
@@ -31,26 +32,92 @@ type VersionRange =
     static member BasicOperators = ["~>";"==";"<=";">=";"=";">";"<"]
     static member StrategyOperators = ['!';'@']
     static member Exactly version = Specific(SemVer.Parse version)
-
-    static member Between(version1,version2) = Range(VersionRangeBound.Including, SemVer.Parse version1, SemVer.Parse version2, VersionRangeBound.Excluding)
-
+    
+    static member Between(lower,minimum,maximum,upper) =
+        Range(lower,minimum,maximum,upper)      
+
+    static member Between(lower,version1,version2,upper) =
+        let minimum = SemVer.Parse version1
+        let maximum = SemVer.Parse version2
+        VersionRange.Between(lower,minimum,maximum,upper)
+        
+    static member Between(minimum:string,maximum:string) =
+        VersionRange.Between(VersionRangeBound.Including,minimum,maximum,VersionRangeBound.Excluding)
+              
+    static member Between(minimum:SemVerInfo,maximum:SemVerInfo) =
+        VersionRange.Between(VersionRangeBound.Including,minimum,maximum,VersionRangeBound.Excluding)
+        
     member x.IsGlobalOverride = match x with | OverrideAll _ -> true | _ -> false
 
     member this.IsIncludedIn (other : VersionRange) =
         match other, this with
         | Minimum v1, Minimum v2 when v1 <= v2 -> true
         | Minimum v1, Specific v2 when v1 <= v2 -> true
+        | Minimum v1, Range(_, min2, max2, _) when v1 <= min2 && v1 <= max2 -> true
         | Specific v1, Specific v2 when v1 = v2 -> true
-        | Range(_, min1, max1, _), Specific v2 when min1 <= v2 && max1 >= v2 -> true
         | GreaterThan v1, GreaterThan v2 when v1 < v2 -> true
         | GreaterThan v1, Specific v2 when v1 < v2 -> true
+        | GreaterThan v1, Range(_, min2, max2, _) when v1 < min2 && v1 < max2 -> true
+        | Range(lower, min1, max1, upper), Specific v2 -> 
```

Comment:
covering more cases allows faster resolver passes, and may prevent looping in some cases